Project Chat Bot: Hannah and Ylva

jesinkey opened the pull request:
Hannah Jesinkey and Ylva Landoff Lindberg

jesinkey committed:
Added User answer

jesinkey committed:
Working Doc.Bot, but with dialogue needed

jesinkey committed:
Working else-function

jesinkey committed:
Working setTimeout

jesinkey committed:
Working 1,2,3,

jesinkey committed:
Adjusted Bot Image

jesinkey committed:
Declared functions at top of page

jesinkey committed:
FINAL

jesinkey committed:
with link in README

KarolCislowski commented on code/script.js:
```diff
@@ -1,47 +1,105 @@
 
-// Functions declared here
+// Functions declared here.
+const userReply = () => {
+    showMessage(userInput.value, 'user')
+}
+const botReply = (message) => {
+    showMessage(message, 'bot')
+}
```
you could add setTimeout here, what would make your code much cleaner and shorter 😸

jesinkey replied:
I don't quite understand?  Can you clarify a bit more?

KarolCislowski replied:
``` 
const botReply = (message, time) => {
    setTimeout(showMessage(message, 'bot'), time)
}
```
This would save you a lot o typing...

jesinkey replied:
Ah OK! we fixed that in the new push

KarolCislowski commented on code/script.js:
```diff
@@ -1,47 +1,105 @@
+        }
+
+        if (userInput.value === `1`) {
+            setTimeout(botReply, 1000, [`So ${name}, I see our time is up. Keep on breathing and I see you next week`])
```
I don't understand why you put your strings inside the array here❓

jesinkey replied:
It was just to make sure that a string was being compared to a string. As I understand it for 3*= it is necessary that the two sides are the same types. And when using the .value is always returns a string, I think?

KarolCislowski commented on code/script.js:
```diff
@@ -1,47 +1,105 @@
+        userReply();
+
+        botReply(`Hmmm... I’ m listening. Please tell me more`)
+    } else if (questionCounter === 3) {
+        userReply();
+
+        setTimeout(botReply, 1000, [`I see! No one really ever asks me how I feel...`])
+        setTimeout(botReply, 4000, [`Sometimes I'm very low on energy. I have alot to process you know. It makes me stressed and affects my mood.`])
+        setTimeout(botReply, 5000, [`Sleep is very important. So are hugs.`])
+        setTimeout(botReply, 8000, [`Please hug me!`])
+        setTimeout(botReply, 11000, [`That feels better, don’t you think?`])
+        setTimeout(botReply, 13000, [`So back to you. How would you describe your wellbeing on a scale of 1 to 3 where 1 is “Like Shit” and 3 is “On Top Of The World”?`])
+    } else {
+        userReply();
+
+        if (userInput.value === `1`) {
+            botReply(`Take a deep breath, then you should probably contact your Vårdcentral and get some more professional help. There is help out there!`)
+        } else if (userInput.value === `2`) {
+            botReply(`Make sure to create balance in your life. Try to find time to exercise and get some daylight every day, but also time to slow down and recharge your battery. Try to be social but take it easy with alcohol, it messes up your sleep.`)
+        } else if (userInput.value === `3`) {
+            botReply(`That’s amazing!`)
+        } else {
+            botReply(`I am afraid I don't understand you answer, please try again or contact another health professional`)
+        }
+
+        if (userInput.value === `1`) {
+            setTimeout(botReply, 1000, [`So ${name}, I see our time is up. Keep on breathing and I see you next week`])
+        } else if (userInput.value === `2`) {
+            setTimeout(botReply, 1000, [`So ${name}, I see our time is up. Hang in there and I see you next week`])
+        } else {
+            setTimeout(botReply, 1000, [`So ${name}, I see our time is up. Nice talking to you! Maybe we can go out parting some day!?`])
+        }
+
+        setTimeout(botReply, 3000, [`Oh, by the way. I’m sending you the bill to your IP address. Bye bye!`])
+    }
+
+    console.log(userInput.value)
+    userInput.value = '';
+    questionCounter = questionCounter + 1;
+});
```
This is a very long function. Please try to cut it on the smaller functions. 🔪

jesinkey replied:
Hi! We had trouble doing this as this was a single conversation and didn't know where to cut it up?

KarolCislowski commented on code/style.css:
```diff
@@ -1,140 +1,149 @@
+    font-size: 24px;
+    line-height: 34px;
+    color: white;
+    ;
```
This semicolon can mess with your styling 🤓

KarolCislowski reviewed (approved):
Hi, I'm Karol Cislowski and I'm Technigo's teacher assistant. Sometimes it will be me providing Code Reviews for your assignments. If you have any questions or want to chat about code then feel free to reach me on Technigo's Slack. So let's come back to our business 😸 

I can see that you had a lot of fun with it and it is the most important 👍🏻 
Try to experiment some more with your code! Don't be scared to break it, because repairing is the best way to learn how something works 🔨 Below I provided some tips about what you can do with your code, or apply in the future projects to rise it to the next level 🚀 

I can wait to see your next projects!

jesinkey committed:
with edits from Code Review

jesinkey commented:
Hi! We took your advice on the setTimeout and changed the botReply function. We could not think of how to cut up the eventListener with the conversation as it is one continuos if-string. We added comments for better clarity. Let us know what you think!
